core/Symbols.h

```diff
@@ -217,56 +217,117 @@ class Field final {
+        bool isInvariant : 1;
+        bool isContravariant : 1;
+        bool isFixed : 1;
 
-        // --- For our current symbol type, what flags does it have?
+        constexpr static uint8_t NUMBER_OF_FLAGS = 6;
+        constexpr static uint8_t VALID_BITS_MASK = (1 << NUMBER_OF_FLAGS) - 1;
 
-        // Class flags
-        static constexpr uint32_t CLASS_OR_MODULE_CLASS = 0x0000'0010;
-        static constexpr uint32_t CLASS_OR_MODULE_MODULE = 0x0000'0020;
-        static constexpr uint32_t CLASS_OR_MODULE_ABSTRACT = 0x0000'0040;
-        static constexpr uint32_t CLASS_OR_MODULE_INTERFACE = 0x0000'0080;
-        static constexpr uint32_t CLASS_OR_MODULE_LINEARIZATION_COMPUTED = 0x0000'0100;
-        static constexpr uint32_t CLASS_OR_MODULE_FINAL = 0x0000'0200;
-        static constexpr uint32_t CLASS_OR_MODULE_SEALED = 0x0000'0400;
-        static constexpr uint32_t CLASS_OR_MODULE_PRIVATE = 0x0000'0800;
+        Flags() noexcept
+            : isTypeArgument(false), isTypeMember(false), isCovariant(false), isInvariant(false),
+              isContravariant(false), isFixed(false) {}
 
-        // Type flags
-        static constexpr uint32_t TYPE_COVARIANT = 0x0000'0010;
-        static constexpr uint32_t TYPE_INVARIANT = 0x0000'0020;
-        static constexpr uint32_t TYPE_CONTRAVARIANT = 0x0000'0040;
-        static constexpr uint32_t TYPE_FIXED = 0x0000'0080;
+        uint8_t serialize() const {
+            // Can replace this with std::bit_cast in C++20
+            auto rawBits = *reinterpret_cast<const uint8_t *>(this);
+            // Mask the valid bits since uninitialized bits can be any value.
+            return rawBits & VALID_BITS_MASK;
+        }
+
+        bool hasFlags(const Flags other) {
+            const auto otherSerialized = other.serialize();
+            return (this->serialize() & otherSerialized) == otherSerialized;
+        }
+    };
+    CheckSize(Flags, 1, 1);
+
+    Loc loc() const;
+    const InlinedVector<Loc, 2> &locs() const;
+    void addLoc(const core::GlobalState &gs, core::Loc loc);
+
+    uint32_t hash(const GlobalState &gs) const;
+
+    bool isPrintable(const GlobalState &gs) const;
+
+    SymbolRef ref(const GlobalState &gs) const;
+
+    Variance variance() const {
+        if (flags.isInvariant) {
+            return Variance::Invariant;
+        }
+        if (flags.isCovariant) {
+            return Variance::CoVariant;
+        }
+        if (flags.isContravariant) {
+            return Variance::ContraVariant;
+        }
+        Exception::raise("Should not happen");
+    }
+
+    SymbolRef dealias(const GlobalState &gs, int depthLimit = 42) const;
+
+    void sanityCheck(const GlobalState &gs) const;
+
+    TypeParameter deepCopy(const GlobalState &gs) const;
+
+    Flags flags;
+    // Method for TypeArgument, ClassOrModule for TypeMember.
+    SymbolRef owner;
+    NameRef name;
+    TypePtr resultType;
+
+private:
+    InlinedVector<Loc, 2> locs_;
+};
+
+class ClassOrModule final {
```

Comment:
Historically `Symbol` couldn't have `CheckSize` because `UnorderedMap` has different sizes in different environments. I dunno if that is still true.

Reply:
The assertions from `CheckSize` are only active for `!debug_mode && !emscripten_build`...so I guess there'd have to be significant differences between x86-64 Linux and Mac (which doesn't seem likely)  to cause problems?  Maybe there would be latent issues for aarch64?